### ISSUE TITLE
[Job] "no ray driver" when submitted ray job to headnode 

### ISSUE THREAD

**Issue by AndreKuu** [ray-project/ray#40443]
### What happened + What you expected to happen

I submitted some ray job to headnode. I found some jobs in "no ray driver" infinty. I found some logs in the python-core-driver.log. Here there are:
"dashboard_agent.log:49809:2023-10-18 04:45:04,580       INFO job_manager.py:879 -- Starting job with submission_id: 263212545118446096
events/event_JOBS.log:57:{"event_id": "22b04A059aeEACB4e4cD8101aDDD5FABB4CF", "source_type": "JOBS", "source_hostname": "apa-prod-raycluster-nbg9f-head-ms72f", "source_pid": 193, "message": "Started a ray job 263212545118446096.", "timestamp": "1697604304", "custom_fields": {"submission_id": "263212545118446096"}, "severity": "INFO", "label": ""}
Binary file gcs_server.out matches
python-core-driver-01000000ffffffffffffffffffffffffffffffffffffffffffffffff_193.log:20999:[2023-10-18 04:45:04,594 W 193 193] actor_manager.cc:112: Failed to look up actor with name '_ray_internal_job_actor_263212545118446096'. This could because 1. You are trying to look up a named actor you didn't create. 2. The named actor died. 3. You did not use a namespace matching the namespace of the actor.
python-core-driver-01000000ffffffffffffffffffffffffffffffffffffffffffffffff_193.log:21054:[2023-10-18 04:45:52,155 I 193 231] task_manager.cc:800: Task failed: SchedulingCancelled: Actor creation cancelled.: Type=ACTOR_CREATION_TASK, Language=PYTHON, Resources: {}, function_descriptor={type=PythonFunctionDescriptor, module_name=ray.dashboard.modules.job.job_manager, class_name=JobSupervisor, function_name=__init__, function_hash=397a8101a9524dd68f0e1c899d9cc7bd}, task_id=ffffffffffffffff6f87a64af727fa848eb5b6f301000000, task_name=_ray_internal_job_actor_263212545118446096:JobSupervisor.__init__, job_id=01000000, num_args=8, num_returns=1, depth=1, attempt_number=0, actor_creation_task_spec={actor_id=6f87a64af727fa848eb5b6f301000000, max_restarts=0, max_retries=0, max_concurrency=1000, is_asyncio_actor=1, is_detached=1}, runtime_env_hash=-2092952565, eager_install=1, setup_timeout_seconds=600"

It said that the job supervisor actor init failed because of "SchedulingCancelled". It looks like that it caused by some system failures and system call the actor.kill() ?

May I ask if any guys can help me take a look about this issue?
![image](https://github.com/ray-project/ray/assets/76543919/470e1485-0f77-4cf4-b1c5-786d5c3c5059)

### Versions / Dependencies

2.5.0

### Reproduction script

Just use the job API to submitt a job. It is easier to reproduce when a head node has completed more jobs in total.

### Issue Severity

High: It blocks me from completing my task.

**Comment by anyscalesam** [comment]
@scottsun94 can you please triage and set priority assign

**Comment by scottsun94** [comment]
@AndreKuu For the jobs with "no ray driver", did you use ray or have a ray driver in your entrypoint script?

**Comment by AndreKuu** [comment]
> n your entrypoint script?

yes, i'm sure using ray in the job entrypoint. All jobs have the same entrypoint, the other jobs work well.

**Comment by AndreKuu** [comment]
> @AndreKuu For the jobs with "no ray driver", did you use ray or have a ray driver in your entrypoint script?

There some ray related code in entrypoint script:
```python
import ray
ray.init(runtime_env=runtime_env, log_to_driver=False)
...
```

**Comment by anyscalesam** [comment]
@scottsun94 please take the next step here

**Comment by scottsun94** [comment]
I see. This is not a dashboard issue then.
Need @akshay-anyscale to triage

**Comment by anyscalesam** [comment]
@sihanwang41 please triage

**Comment by shrekris-anyscale** [comment]
Could you try this out on Ray 2.7.1, and let us know if that fixes the issue?

**Comment by nhha1602** [comment]
I got the same issue with simple command as below:

`ray job submit --address $RAY_ADDRESS --runtime-env-json='{"pip": ["requests==2.26.0"]}' --working-dir ./ -- python test_runtime_env.py`

I tried to test this because my ray cluster could not go to internet for installing packages.

So, how can I stop/delete this summit ? because I did not see JOB_ID
Or I have to delete cluster ?

Thanks and regards

**Comment by LenzoAB** [comment]
Hello!
Any update on this ticket? We are having the same trouble on our end. Thank you!

**Comment by AndreKuu** [comment]
您的邮件已收到，祝好。------KuU
